packages/terminal/src/common/shell-terminal-protocol.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+export const IShellTerminalServer = Symbol('IShellTerminalServer');
+
+export const shellTerminalPath = '/services/shell-terminal';
+
+export interface IShellTerminalServerOptions {
```

Comment:
Would it be possible to add `IShellTerminalServer` which extends `ITerminalServer` and overrides `create` method to specialize options.

Reply:
Yes actually I think I could do that, good point.

Reply:
I just remembered why I could not do that...

The problem is that if let's say I have:

```
export interface ITerminalServer extends JsonRpcServer<ITerminalClient> {
    create(options: ITermianlServerOptions): Promise<number>;
    resize(id: number, cols: number, rows: number): Promise<void>;
}
```
Then I can't extend this interface and override create.

And if I have the interface without the create method, then I can extend it but I can't use the create method so :(